### PR TITLE
(EAI-576) [UI] Add React Testing Library + tests

### PR DESCRIPTION
**Pull request by nlarew**
Jira: (EAI-576) [UI] Add React Testing Library + tests

## Changes

- Adds RTL and a test for `useConversation()` (the hook with most chatbot logic, others consume this)

## Notes

- We still can't test the chatbot's open behavior due to a bug with react-markdown. Need to find a workaround for this https://jira.mongodb.org/browse/EAI-589

### CODE REVIEW

File: packages/mongodb-chatbot-ui/src/useConversation.tsx [hunk 1]
```diff
@@ -209,6 +209,10 @@ function conversationReducer(
         );
         return state;
       }
+      console.log(`Deleting message with id ${action.messageId}`, [
+        ...state.messages.slice(0, messageIndex),
+        ...state.messages.slice(messageIndex + 1),
+      ]);
```
Comment (mongodben) [inline packages/mongodb-chatbot-ui/src/useConversation.tsx:215]: just confirming this console.log should be present in the code

File: packages/mongodb-chatbot-ui/tsconfig.json [hunk 2]
```diff
@@ -18,7 +20,9 @@
     "noFallthroughCasesInSwitch": true,
     "noImplicitAny": true,
     "noImplicitReturns": true,
-    "noImplicitThis": true
+    "noImplicitThis": true,
+
+    "types": ["vitest/globals", "@testing-library/jest-dom"]
   },
-  "include": ["./src/**/*.ts", "./src/**/*.tsx"]
+  "include": ["./src/**/*.ts", "./src/**/*.tsx", "./vitest.setup.ts"],
```
Comment (mongodben) [inline packages/mongodb-chatbot-ui/tsconfig.json:27]: is this saying that test set up should be included in the bundle?

**Review by mongodben — APPROVED** [review]
lgtm, few small things and Qs

File: packages/mongodb-chatbot-ui/tsconfig.json (hunk 2 above)
Comment (nlarew) [reply packages/mongodb-chatbot-ui/tsconfig.json]: Yeah it is - not sure why I included it but I don't think it's necessary

**Pull request closed without merging**